pkg/probe/probe.go

```diff
@@ -41,19 +44,21 @@ type Signal struct {
 }
 
 type prober struct {
-	endpoint  string
-	fqdn      string
-	port      string
-	addresses []net.IP
-	conn      net.Conn
-	tlsConn   *tls.Conn
-	sni       string
+	endpoint          string
+	fqdn              string
+	port              string
+	addresses         []net.IP
+	conn              net.Conn
+	tlsConn           *tls.Conn
+	sendProxyProtocol string
+	sni               string
 }
 
 func NewProber(o ProbeOptions) (*prober, error) {
 	p := &prober{
-		endpoint: o.Endpoint,
-		sni:      o.ServerNameIndication,
+		endpoint:          o.Endpoint,
+		sendProxyProtocol: o.ProxyProtocol,
```

Comment:
sendProxyProtocol sounds like a boolean, but you actually pass v1/v2 there

Reply:
Fixed by making it an enum for the "mode".